src/include/lexer/lexer.h

```diff
@@ -76,6 +80,23 @@ class Lexer {
    * during tokenization.
    */
   double floatval_;
```

Comment:
Is there a specific reason all variables are public?
Most are not being accessed from other classes.
Use getters where applicable.

Reply:
It is used for in testing. 
Changing it would require huge changes, but ok will try

Reply:
used for testing is not a valid reason

Reply:
make private

Reply:
Moved it out of public , should be default to private. 👀